### ISSUE TITLE
Option to queue folder after curren folder is finished, not after track

### ISSUE THREAD

**Issue by legion1978** [Taiko2k/TauonMusicBox#27]
Hello. This is more like a feature request--
It would be nice to let an album finish before starting the next one. Right now if it's in the middle of it, it will just skip the remaining tracks and jump right to next album. The option would be "finish album currrntly playing, then jump to next one"
thanks!

**Comment by Taiko2k** [comment]
This feature partly currently exists; if you add the current album to the queue first, it will finish rather than repeating. 

I could add in right click queue menu a function `Finish Current Album`. This would save having to find the original album first. Would also make this feature better exposed.

**Comment by Taiko2k** [comment]
Though.. you do need to have the queue panel open for the above to be convenient. I could perhaps also add in addition to `Add Album to Queue` something like `...To Queue and finish Current` to the track folder menu.

**Comment by Taiko2k** [comment]
I've pushed a release with both the function in the queue right click menu, and an option in settings under `Settings` > `Playlist` > `Finish current when queuing album`

Hopefully this works well. Let me know if there are any issues.

**Comment by legion1978** [comment]
hello. i just noticed that for some reason Tauon is not honoring the "finish current album when queueing" setting. it jumps immediatly to the queued album after playing a track. im pretty sure it was working before... mi using version 3.6.0

**Comment by Taiko2k** [comment]
Ah yes you're right. Seems to be broken when wheel clicking, but not when using context menu. I'll fix this.

I also noticed another bug where it adds multiple part way albums when adding more to the queue, ill fix that too.

**Comment by legion1978** [comment]
excellent

**Comment by legion1978** [comment]
you know i was thinking. once you know how the queue works (middle click vs right click), is not such a bad idea. you could leave it as it is or maybe make it ctrl or shift [album title] click. this provides an additional option right off the bat, with a desired default behavior.

**Comment by Taiko2k** [comment]
Hmm, I think having middle and menu behave differently might be a bit too un-intuitive.  

Though, I think it would be nice to have a way in which the setting is not needed at all. Maybe ctrl+click to finish current. But perhaps that's less convenient than how it works currently?

**Comment by legion1978** [comment]
id say the current behaviour would be ok, provided its documented. i realized it exactly because of the convenience of it. i usually let the current album finish (default setting) but if i just wanted to listen to one track and then jump right to another album, id have to go and modify the setting, do the thing, and later re-set it. whereas the middle click would do just that, no further config altering needed. but, the user must be aware of this behaviour beforehand.

**Comment by Taiko2k** [comment]
OK, I'll leave it how it is then.

How about this idea:
 - Wheel clicking always adds to ***end*** of queue
 - We remove the option from settings
 - Add a 2nd option to menu that always queues the album *after* the current one

So it would look something like this (I'm not sure on what the wording should be):
![screenshot from 2019-02-09 19-33-48](https://user-images.githubusercontent.com/17271572/52517566-031bfa00-2ca2-11e9-98d7-ddaf8cd799c1.png)

Then there's less confusion, we don't need extra documentation, and this function will work even if there are already albums in the queue but we want to *insert* an album next.

**Comment by legion1978** [comment]
"Add a 2nd option to menu that always queues the album after the current one" > after current track or album?

i think it should be "queue album after current track" and "queue album (after current album)", but seems kinda long.

**Comment by Taiko2k** [comment]
Yes that was what I was having difficulty with the text needs to be short.

It would work like this:
 - **Queue album:** Adds to ***end*** of queue (So if nothing else is in the queue, the album will be played immedialy after current track)
 - **Queue album next** - Adds album to ***start*** of queue, but after any in-progress album. If the queue if empty, finish current album

**Comment by Taiko2k** [comment]
Perhaps this wording is more clear:

![screenshot from 2019-02-10 14-05-17](https://user-images.githubusercontent.com/17271572/52528199-352e6a00-2d3d-11e9-8d59-6a5a6a2b59bd.png)

**Comment by legion1978** [comment]
all in all, it should be
| Add album to queue
  > after current track
  > after current album
which i think is unambiguous, but its your call. maybe some users could give some other insights..
